docker/docker-entrypoint.sh

```
@@ -1,410 +1,185 @@
+
+# --- GPG Agent Setup ---
+setup_gpg_agent() {
+    log "--- Setting up GPG Agent ---"
+    # Create the necessary directory for the GPG agent socket
+    mkdir -p /run/user/appuser
+    chown appuser:appuser /run/user/appuser
+    # Start the GPG agent as the 'appuser'
+    # The socket path is explicitly set to be in a predictable location
+    gosu appuser gpg-agent --daemon --pinentry-mode loopback --homedir /home/appuser/.gnupg --allow-extra-socket --log-file /app/logs/gpg-agent.log
+    # Export the GPG TTY to prevent pinentry issues
+    export GPG_TTY=$(tty)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)